docs/02-app/01-building-your-application/02-data-fetching/01-fetching-caching-and-revalidating.mdx

```diff
@@ -0,0 +1,299 @@
+
+You can use `fetch` with [`async`/`await` in Server Components](https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises), in [Route Handlers](/docs/app/building-your-application/routing/route-handlers), and in [Server Actions](/docs/app/building-your-application/data-fetching/server-actions).
+
+For example:
+
+```tsx filename="app/page.tsx" switcher
+async function getData() {
+  const res = await fetch('https://api.example.com/...')
+  // The return value is *not* serialized
+  // You can return Date, Map, Set, etc.
+
+  if (!res.ok) {
+    // This will activate the closest `error.js` Error Boundary
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export default async function Page() {
+  const data = await getData()
+
+  return <main></main>
+}
+```
+
+```jsx filename="app/page.js" switcher
+async function getData() {
+  const res = await fetch('https://api.example.com/...')
+  // The return value is *not* serialized
+  // You can return Date, Map, Set, etc.
+
+  if (!res.ok) {
+    // This will activate the closest `error.js` Error Boundary
+    throw new Error('Failed to fetch data')
+  }
+
+  return res.json()
+}
+
+export default async function Page() {
+  const data = await getData()
+
+  return <main></main>
+}
+```
+
+> **Good to know**:
+>
+> Next.js provides helpful functions you may need when fetching data in Server Components such as [`cookies`](/docs/app/api-reference/functions/cookies) and [`headers`](/docs/app/api-reference/functions/headers). These will cause the route to be dynamically rendered as they rely on request time information.
+> In Route handlers, `fetch` requests are not memoized as Route Handlers are not part of the React component tree.
+> To use `async`/`await` in a Server Component with TypeScript, you'll need to use TypeScript `5.1.3` or higher and `@types/react` `18.2.8` or higher.
+
+### Caching Data
+
+Caching stores data so it doesn't need to be re-fetched from your data source on every request.
+
+By default, Next.js automatically caches the returned values of `fetch` in the [Data Cache](/docs/app/building-your-application/caching#data-cache) on the server. This means that the data can be fetched at build time or request time, cached, and reused on each data request.
+
+```js
+// 'force-cache' is the default, and can be omitted
+fetch('https://...', { cache: 'force-cache' })
+```
+
+`fetch` requests that use the `POST` method are also automatically cached in the Data Cache – unless it's used in `POST` [Route Handler](/docs/app/building-your-application/routing/route-handlers) or comes after `headers`/`cookies`.
```

Comment:
At least current implementation, it is true. 

Also, please refer to https://github.com/vercel/next.js/pull/52669#issuecomment-1638295085 I'm not sure this my comment is right because I have not recieved any response from the Vercel staff...

Reply:
I've grouped every case and asked for clarity on Slack. I think it's a lot more nuanced, and it'll be good to dig into the commerce use case.